rocketc/static/js/src/rocketc.js

```diff
@@ -96,4 +99,29 @@ function RocketChatXBlock(runtime, element) {
         });
     });
 
+    function responseGetGroups(data){
+        $("#select-channel").empty();
+        for (var i in data){
+            var item =  $("<option>"+data[i]+"</option>");
+            $("#select-channel").append(item);
+        };
+        $("#select-channel").change(function(){
+            url = $("#tool-buttons").attr("data-domain") + $("#select-channel").val() +"?layout=embedded";
+            $("#myframe").attr("src", url);
+        });
+    };
+
+    function loadGroups(){
+        data = {"userId": $("#tool-buttons").attr("data-userId"),
+                "authToken": $("#tool-buttons").attr("data-token")};
+        $.ajax({
```

Comment:
why you use a POST request to load groups, shouldn't it be a GET?

Reply:
the decorator @Xblock.json_handler requires a POST